base/common/LocalDateTime.h

```diff
@@ -67,7 +67,7 @@ class LocalDateTime
 
         m_hour = (s[11] - '0') * 10 + (s[12] - '0');
         m_minute = (s[14] - '0') * 10 + (s[15] - '0');
-        m_second = (s[17] - '0') * 10 + (s[18] - '0');
+        m_second = ((s[17] - '0') * 10 + (s[18] - '0')) % 60;
```

Comment:
It's not zero-cost.

Reply:
char is signed on x86, it will be promoted to signed integer; the modulo of division in C++ for negative numbers returns negative number.

Reply:
A comment needed (but better to do something different).

Reply:
>but better to do something different

Let's just forbid negative arguments, as @vdimir suggested

Another option is to make DateTime64 behavior for negative values the same as for DateTime - return start of the epoch (note, that here toDateTime() returns DateTime64, since the second argument is numeric)